src/Common/Common.Authentication/WindowsAuthenticationFactory.cs

```diff
@@ -119,7 +174,16 @@ private static string GetExecutingDirectory()
             var fileUri = new UriBuilder(codeBase);
             var directory = Uri.UnescapeDataString(fileUri.Path);
             directory = Path.GetDirectoryName(directory);
-            return directory;
+            if (string.IsNullOrEmpty(fileUri.Host))
```

Comment:
Code is duplicated. Consider centralizing.

Reply:
Yeah we need a Utility assembly or reuse the abstraction assembly